underdocs-error/src/main/kotlin/underdocs/error/pretty/DefaultErrorPrettyPrinter.kt

```diff
@@ -1,7 +1,34 @@
 package underdocs.error.pretty
 
+import underdocs.error.BaseUnderdocsError
+
 class DefaultErrorPrettyPrinter : ErrorPrettyPrinter {
     override fun print(exception: Exception): String {
```

Comment:
This function does two things. If the `exception` parameter descends from `BaseUnderdocsError`, then it performs a pretty-print with all the details, otherwise defaults to an "unexpected error" mode.

Hence, in my opinion, it would be much cleaner to extracts these functionalities into two separate functions. Then you could also easily turn this into an expression-bodied function:

~~~~Kotlin
override fun print(exception: Exception) = if (exception is BaseUnderdocsException) {
  printBaseUnderdocsException(exception)
} else {
  printUnexpectedException(exception)
}
~~~~

An unfamiliar reader would immediately see our intention: the separate handling of these two cases.

On top of that, we also saved ourselves from lots of syntactical noise: the `return` keywords and the explicit return type. It just feels cleaner, isn't it?
    
See [Single-expression functions](https://kotlinlang.org/docs/reference/functions.html#single-expression-functions).

Reply:
You are right, this reads so much better! 
The two smaller functions I created became single-expression functions as well. Should I keep them as the usual block styled functions or are they just as readable this way?

Reply:
`printUnexpectedException` is great as a single-expression one, however, I'm unsure in the case of `printBaseUnderdocsException`. Please see my comment below.